api/datastore/sql/migrations/old.go

```diff
@@ -0,0 +1,112 @@
+package migrations
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"github.com/fnproject/fn/api/common"
+	"github.com/jmoiron/sqlx"
+	"github.com/pressly/goose"
+)
+
+var oldMigrationsTable = "schema_migrations"
+
+func checkOldMigrationTableVersionIfExists(db *sqlx.DB) (version int64, dirty bool, err error) {
+	ctx := context.Background()
+
+	q := db.QueryRowContext(
+		ctx, "SELECT version, dirty FROM "+oldMigrationsTable+" LIMIT 1")
+	q.Scan(&version, &dirty)
+	if err == sql.ErrNoRows {
+		return -1, false, nil
+	} else if err != nil {
+		return -1, false, err
+	}
+	return version, dirty, nil
+}
+
+func checkOldMigration(ctx context.Context, db *sqlx.DB) (int64, error) {
+	log := common.Logger(ctx)
+	current, dirty, err := checkOldMigrationTableVersionIfExists(db)
+	if err != nil {
+		return -1, err
+	}
+	if dirty {
+		log.Fatal("database corrupted")
+	}
+	if current > 0 {
+		// will cause partial upgrade, starting from
+		// the latest version in the old migration table
+		return current, nil
+	}
+	// will cause full upgrade
+	return -1, nil
+}
+
+func syncToGooseTable(ctx context.Context, db *sqlx.DB) (goose.Migrations, error) {
+	log := common.Logger(ctx)
+	goose.SetDialect(db.DriverName())
+
+	// this will create goose migrations table if it doesn't exist
+	gooseCurrent, err := goose.GetDBVersion(db.DB)
+	if err != nil {
+		return nil, err
+	}
+	log.Debug("current goose migrations version: ", gooseCurrent)
+
+	// will return the latest version
+	migrateCurrent, err := checkOldMigration(ctx, db)
+	if err != nil {
+		return nil, err
+	}
+	log.Debug("current mattes migrations version: ", migrateCurrent)
+
+	sortedMigrations := sortAndConnectMigrations(migrations)
+	newMigrationsQuery := fmt.Sprintf("INSERT INTO goose_db_version (version_id, is_applied) VALUES (?, ?);")
```

Comment:
postgres uses `$` syntax and oracle db will use `:named` syntax for these queries

Reply:
all these statements going to rebind, so it works fine for all our datastores.